rubrics/api/src/main/java/org/sakaiproject/rubrics/logic/RubricsService.java

```diff
@@ -62,4 +62,8 @@ void saveRubricEvaluation(String toolId,
     String generateLang();
 
     String getRubricEvaluationObjectId(String associationId, String userId, String toolId);
+
+    void deleteRubricAssociation(String query, String toolId);
+
+    void deleteSamigoAssociatedResources(String itemId);
```

Comment:
yeah api for a specific tool is not ideal need to rethink this.

Reply:
Maybe I could rename it and call it something like deleteAssociationUsingWildCards? This is specific to Samigo and just for improving performance (otherwise it's not needed). If an assessment is deleted, we don't know if it has 1 or 100 items and if each of them has a rubric association or not, using this method we're encapsulating it all in a single service call, saving many rest calls.

Reply:
Completely agree on the approach just needed to be generalized.

I don't like "Wildcard", how about:
How about `deleteRubricAssociationsByItemId`
and
`Traverson.TraversalBuilder builder = traverson.follow("rubric-associations", "search", "by-item-id");`

Reply:
@ern

Reply:
yep fine with that